cdap-master/src/main/java/co/cask/cdap/data/tools/DataMigration.java

```diff
@@ -0,0 +1,250 @@
+                                   UpgradeMetricsConstants.DEFAULT_METRICS_TABLE_PREFIX) + ".agg";
+
+    // versions older than 2.7, has two metrics table, identified by system and user prefix
+    String tableName26 = "system." + tableName27;
+    DefaultDatasetNamespace defaultDatasetNamespace = new DefaultDatasetNamespace(cConf);
+    String metricsEntityTable26 = defaultDatasetNamespace.namespace(new Id.Namespace(Constants.SYSTEM_NAMESPACE),
+                                                                    tableName26);
+    String metricsEntityTable27 = defaultDatasetNamespace.namespace(new Id.Namespace(Constants.SYSTEM_NAMESPACE),
+                                                                    tableName27);
+    HBaseAdmin hAdmin = null;
+    Version version = null;
+    try {
+      hAdmin = new HBaseAdmin(hConf);
+      for (HTableDescriptor desc : hAdmin.listTables()) {
+        if (desc.getNameAsString().equals(metricsEntityTable27)) {
+          System.out.println("Matched HBase Table Name For Migration " + desc.getNameAsString());
+          version = metricHBaseTableUtil.getVersion(desc);
+          version = verifyVersion(Version.VERSION_2_7, version);
+          if (version == null) {
+            return null;
+          }
+          break;
+        }
+        if (desc.getNameAsString().equals(metricsEntityTable26)) {
+          System.out.println("Matched HBase Table Name For Migration " + desc.getNameAsString());
+          version = metricHBaseTableUtil.getVersion(desc);
+          version = verifyVersion(Version.VERSION_2_6_OR_LOWER, version);
+          if (version == null) {
+            return null;
+          }
+        }
+      }
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+    return version;
+  }
+
+  private Version verifyVersion(Version expected, Version actual) {
+    if (expected != actual) {
+      System.out.println("Version detected based on table name does not match table configuration");
+      return null;
+    }
+    return actual;
+  }
+
+  public static void main(String[] args) throws Exception {
+    new DataMigration().doMain(args);
+  }
+
+  /**
+   * Sets up a {@link DatasetFramework} instance for standalone usage.  NOTE: should NOT be used by applications!!!
+   */
+  public static DatasetFramework createRegisteredDatasetFramework(Injector injector)
+    throws DatasetManagementException, IOException {
+    DatasetDefinitionRegistryFactory registryFactory = injector.getInstance(DatasetDefinitionRegistryFactory.class);
+    DatasetFramework datasetFramework =
+      new InMemoryDatasetFramework(registryFactory, injector.getInstance(CConfiguration.class));
+    // TODO: this doesn't sound right. find out why its needed.
+    datasetFramework.addModule(Id.DatasetModule.from(Constants.SYSTEM_NAMESPACE, "table"),
+                               new HBaseTableModule());
+    datasetFramework.addModule(Id.DatasetModule.from(Constants.SYSTEM_NAMESPACE, "metricsTable"),
```

Comment:
but if hard to figure out why - let's keep for now all and leave the todo - as you did